### PR TITLE
[#48] [UI] As a user, I can see a text-area question and refactor SurveyBoldText to be reusable

### LINKED ISSUE #48: [UI] As a user, I can see a text-area question
## Why
A text-area question type is a question type that allows a user to fill in a multi-line text input. It is commonly used when we would like to ask an open-end question, such as asking for suggestions or comments.

**Example questions**
- _Do you have any suggestions?_
- _Your comments are appreciated!_

## Acceptance Criteria
For text-area questions,
- Display a textbox that supports multi-line text input
  - Display a placeholder inside the textbox
  - The textbox is scrollable when the number of lines goes beyond its viewport (i.e., when the text input no longer fits within the textbox)
- Display the keyboard when a textbox is selected
  - Tap anywhere outside of textbox to release the keyboard
  - The keyboard must not cover what the user is typing

Image

## Design
[Text-area Question](https://www.figma.com/file/GjRPOjDyZ6f4EDL3wKarRK/Challenge-Mobile-App?node-id=2632%3A674)

### PR DESCRIPTION
**Pull request by kaungkhantsoe**
Closes #48 

## What happened 👀

A text-area question type is a question type that allows a user to fill in a multi-line text input. It is commonly used when we would like to ask an open-end question, such as asking for suggestions or comments.

**Example questions**
- _Do you have any suggestions?_
- _Your comments are appreciated!_

## Insight 📝

For text-area questions,
- Display a textbox that supports multi-line text input
  - Display a placeholder inside the textbox
  - The textbox is scrollable when the number of lines goes beyond its viewport (i.e., when the text input no longer fits within the textbox)
- Display the keyboard when a textbox is selected
  - Tap anywhere outside of textbox to release the keyboard
  - The keyboard must not cover what the user is typing

**_Fixed Bugs_**
- There is a bug in zoom in and zoom out feature which cause the `Coroutine Thread` to be cancelled. Fixed it by calling `popBack` function of the navigation instead of calling the function with zoom out animation.
- Flickering effect during zoom in and zoom out is solved by wrapping the `nav host` inside `Surface`

## Proof Of Work 📹

https://user-images.githubusercontent.com/32578035/196885191-9962fc3c-b856-42e9-ae70-62fb3da380e1.mp4

### CODE REVIEW

File: app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/survey/SurveyTextAreaQuestionScreen.kt [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        value = answer,
+        onValueChange = {
+            answer = it
+        },
+        shape = RoundedCornerShape(10.dp),
+        placeholder = {
+            SurveyText(
+                text = stringResource(id = R.string.survey_question_your_thoughts),
+                fontSize = 17.sp,
+                color = White30
```
Comment (ryan-conway) [inline app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/survey/SurveyTextAreaQuestionScreen.kt]: I understand that we must follow the Figma design, but the text is barely legible on complex backgrounds 👀 Shall we increase the opacity, what do you think? 🤔 
CC: @AVI5HEK

File: app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/common/SurveyText.kt [hunk 2]
```diff
@@ -11,19 +11,20 @@ import androidx.compose.ui.unit.TextUnit
 import com.kks.nimblesurveyjetpackcompose.ui.theme.NeuzeitFamily
 
 @Composable
-fun SurveyBoldText(
+fun SurveyText(
```
Comment (ryan-conway) [inline app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/common/SurveyText.kt:16]: Seems like this change should have been mentioned in the PR description, it seems pretty major since many files are affected 🤔

File: app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/home/HomeScreen.kt [hunk 3]
```diff
@@ -182,12 +182,12 @@ fun SurveyContent(
             },
             verticalArrangement = Arrangement.spacedBy(4.dp)
         ) {
-            SurveyText(
+            SurveyCrossfadeText(
```
Comment (ryan-conway) [inline app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/home/HomeScreen.kt:185]: This change also seems out of scope for this PR 🤔

File: app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/survey/SurveyQuestionScreen.kt [hunk 4]
```diff
@@ -1,44 +1,63 @@
...
 @Composable
 fun SurveyQuestionScreen(
     surveyQuestion: SurveyQuestion,
     pageNumber: Int,
     totalNumberOfPage: Int,
     onChooseAnswer: (questionId: String, surveyAnswers: List<SurveyAnswer>) -> Unit
 ) {
+    val keyboardController = LocalSoftwareKeyboardController.current
+    val focusManager = LocalFocusManager.current
+    val coroutineScope = rememberCoroutineScope()
```
Comment (ryan-conway) [inline app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/survey/SurveyQuestionScreen.kt]: Shall we apply this to the `Box` in `SurveyDetailScreen` instead? Tapping on the area above the question (for example, to the right of the `(x)` button) doesn't remove the focus 👀

File: app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/survey/SurveyTextAreaQuestionScreen.kt [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+    TextField(
+        value = answer,
+        onValueChange = {
+            answer = it
+        },
+        shape = RoundedCornerShape(10.dp),
+        placeholder = {
+            SurveyText(
+                text = stringResource(id = R.string.survey_question_your_thoughts),
```
Comment (ryan-conway) [inline app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/survey/SurveyTextAreaQuestionScreen.kt]: ~~Shall we use the `short_text` property of the question here? Seems more appropriate than a hardcoded value 👀~~ 

Screenshot 2565-10-20 at 15 31 03

**Review by ryan-conway — COMMENTED** [review]
A few concerns 👀

File: app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/home/HomeScreen.kt (hunk 3 above)
Comment (kaungkhantsoe) [reply app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/home/HomeScreen.kt:185]: I wanted to change the name of `SurveyBoldText` to `SurveyText` 😬 . So I needed to change this name to a different name.

File: app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/common/SurveyText.kt (hunk 2 above)
Comment (kaungkhantsoe) [reply app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/common/SurveyText.kt:16]: Updated

File: app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/survey/SurveyTextAreaQuestionScreen.kt [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    shortText: String,
+    onChooseAnswer: (answers: List<SurveyAnswer>) -> Unit
+) {
+    var answer by remember { mutableStateOf("") }
+
+    TextField(
+        value = answer,
+        onValueChange = {
+            answer = it
+        },
```
Comment (ryan-conway) [inline app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/survey/SurveyTextAreaQuestionScreen.kt]: We could one-line this :eyes:
```suggestion
        onValueChange = { answer = it },
```

**Review by ryan-conway — COMMENTED** [review]
One minor comment, the rest LGTM 👍🏻

File: app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/survey/SurveyTextAreaQuestionScreen.kt (hunk 5 above)
Comment (ryan-conway) [reply app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/survey/SurveyTextAreaQuestionScreen.kt]: @kaungkhantsoe My mistake, it seems like the hint should be based on `Answer.text`, similar to the Text Field questions 🙈

Comment (kaungkhantsoe) [reply app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/survey/SurveyTextAreaQuestionScreen.kt]: @ryan-conway I will update this in integrate PR since `SurveyTextAreaQuestionScreen` is not really used for now

**Review by ryan-conway — APPROVED** [review]
💪🏻

**Review by AVI5HEK — APPROVED** [review]
